src/subdomains/generic/user/services/webhook/webhook.service.ts

```diff
@@ -0,0 +1,190 @@
+import { BuyFiat } from 'src/subdomains/core/sell-crypto/buy-fiat/buy-fiat.entity';
+import { BuyCrypto } from 'src/subdomains/core/buy-crypto/process/entities/buy-crypto.entity';
+import { TradingLimit } from '../../models/user/dto/user.dto';
+import { WalletService } from '../../models/wallet/wallet.service';
+import { UserRepository } from '../../models/user/user.repository';
+import { SpiderDataRepository } from '../../models/spider-data/spider-data.repository';
+import { KycCompleted, KycStatus, UserData } from '../../models/user-data/user-data.entity';
+
+export enum KycWebhookStatus {
+  NA = 'NA',
+  LIGHT = 'Light',
+  FULL = 'Full',
+  REJECTED = 'Rejected',
+}
+
+export enum KycWebhookResult {
+  STATUS_CHANGED = 'StatusChanged',
+  FAILED = 'Failed',
+}
+
+export enum PaymentWebhookType {
+  BUY_CRYPTO = 'BuyCrypto',
+  BUY_FIAT = 'BuyFiat',
+}
+
+export enum PaymentWebhookState {
+  CREATED = 'Created',
+  COMPLETED = 'Completed',
+}
+
+export class KycWebhookDataDto {
+  mail: string;
+  firstName: string;
+  lastName: string;
+  street: string;
+  houseNumber: string;
+  city: string;
+  zip: string;
+  phone: string;
+  kycStatus: KycWebhookStatus;
+  kycHash: string;
+  tradingLimit: TradingLimit;
+}
+
+export class KycWebhookDto {
+  id?: string;
+  result: KycWebhookResult;
+  data?: KycWebhookDataDto;
+  reason?: string;
+}
+
+export class PaymentWebhookDto {
+  id?: string;
+  type: PaymentWebhookType;
+  state: PaymentWebhookState;
+  inputAmount: number;
+  inputAsset: string;
+  outputAmount: number;
+  outputAsset: string;
+  paymentReference: string;
+}
+
+@Injectable()
+export class WebhookService {
+  constructor(
+    private readonly http: HttpService,
+    private readonly walletService: WalletService,
+    private readonly userRepo: UserRepository,
+    private readonly spiderRepo: SpiderDataRepository,
+    private readonly notificationService: NotificationService,
+  ) {}
+
+  async kycChanged(userData: UserData): Promise<void> {
+    await this.triggerWebhook(userData, await this.getKycWebhookData(userData, KycWebhookResult.STATUS_CHANGED));
+  }
+
+  async kycFailed(userData: UserData, reason: string): Promise<void> {
+    await this.triggerWebhook(userData, await this.getKycWebhookData(userData, KycWebhookResult.FAILED, reason));
+  }
+
+  async paymentUpdate(userData: UserData, payment: BuyFiat | BuyCrypto, state: PaymentWebhookState): Promise<void> {
+    await this.triggerWebhook(userData, this.getPaymentWebhookData(payment, state));
+  }
+
+  // --- HELPER METHODS --- //
+
+  private async triggerWebhook(userData: UserData, data: PaymentWebhookDto | KycWebhookDto): Promise<void> {
+    userData.users = await this.userRepo.find({
+      where: { userData: { id: userData.id } },
+      relations: ['wallet', 'userData'],
+    });
+
+    for (const user of userData.users) {
+      try {
+        const currentUrl = data instanceof KycWebhookDto ? user.wallet.kycUrl : user.wallet.paymentUrl;
```

Comment:
`data instanceof KycWebhookDto` will probably not work, because we are not creating class instances.